### DISCUSSION TITLE
Two questions about the Sharp PC-1403

### DISCUSSION THREAD

**Discussion by ffxx68** [mamedev/discussions#91] [General]
Hi - 

I'm using the PC-1403 emulator right now to debug some programs, and first of all, let me congratulate with Peter Trauner, for the excellent job done, to get these old machines emulated on MESS/MAME. He's the author of the work on the Sharp PC and SC61860 CPU (as per comments in source code):

[pocketc.cpp](https://github.com/mamedev/mame/blob/master/src/mame/sharp/pocketc.cpp)

```
// license:GPL-2.0+
// copyright-holders:Peter Trauner
/******************************************************************************
    Sharp pocket computers
    PC1401/PC1403
    PeT [mess@utanet.at](mailto:mess@utanet.at) May 2000
```
I know a VERY long time has passed, but maybe I'm lucky enough to still get some help here, also because I think it's about general issues with MESS, or MAME, and just partially specific to this machine.

I have two questions:

1) about loading data into device RAM

I tried using the debugger command to load a binary file in memory

`loadr tmp.bin,E0E8,0,:maincpu`

but this doesn't look like loading into the expected locations. 

If I browse the debugger memory window to e0e8, selecting **"Sharp SC61860 ':maincpu' program space"** from the dropdown, I can't find any data uploaded:

![image](https://github.com/mamedev/discussions/assets/659557/b7ad906e-96c1-4b27-9309-014796b5fc31)

but, when I select **"Region ':maincpu'"** from the dropdown, I indeed find the data I expect at E0E8:

![image](https://github.com/mamedev/discussions/assets/659557/f36c985b-8c95-4d4f-987b-bd5fd12fe90b)

I'm sure I'm not specifcying correctly the memory location to upload the file to, but I can't find any way to chose "program space" as the destination.

Can someone help?

2) About moving text, from host to emulated PC

Probably it's more about my ignorance on the MAME system itself, but I'd like to automate the input of relatively long programs on the emulated device.

I read in MESS pages (http://mess.redump.net/mess/howto#pasting_code_to_emulated_computers) that some Copy-Paste is possible, from the host to the emulated device, with the usual Ctrl-c on the host (a Win11 PC) and "S-ScrLock" (Shift-Screen Lock, isn't it?), but I get nothing, apparently. Does it apply to the porting to MAME too? Or, Is there an alternative method? 

I was also thinking about using a LUA script (I understand MAME supports that), to get my text from the host PC being received by the emulated keyboard. But... how? Some examples?

Thanks, for any help!
Fabio (Rome - Italy)

**Comment by cuavas** [comment]
The `loadr` command is used to load into *ROM* regions.  If you want to load into RAM, you need to ensure the system has RAM mapped (if applicable), and then use the `load` command, e.g. `load tmp.bin,E0E8:maincpu,0`.

To paste into the emulated system, you need to ensure UI controls are enabled (Scroll Lock toggles by default), and then press the UI Paste Text key (Shift+Scroll Lock by default).

You can use `emu.keypost(...)` to type text in the emulated system from the Lua console (using the `-console` option).

**Reply by ffxx68** [reply]
Thanks @cuavas for the quick reply !
I tried "load" and indeed data is uploaded to RAM (in spite of a strange I/O error):
![image](https://github.com/mamedev/discussions/assets/659557/734fe614-d5e7-46c3-92e8-6fb85806c17e)
About pasting text ... I'm still not able to do that. And also `emu.keypost(...)` from console doesn't seem to send any character to the device. 
Any suggestion?
Also, is there any way to get in touch with the author, Peter Trauner?

**Reply by cuavas** [reply]
Maybe that driver doesn’t have support for pasting text.
